other/world.ts

```diff
@@ -203,89 +214,95 @@ export default class World {
-        otherBotStats.deaths += 1;
       }
-    }
 
-    // check if bot eats food
-    const food = this.food;
-    const foodIdxToRemove: number[] = [];
-    for (let i = 0; i < food.length; ++i) {
-      // it's safe to cast to Sprite because know that `i` is within the bounds of the array
-      const distance = World.distance(bot, food[i] as Sprite);
-      if (distance < bot.radius) {
-        foodIdxToRemove.push(i);
-        // updating statistic
-        const botStats = this.stats.get(bot.botId);
-        if (!botStats) {
-          console.error(new Error("can't find bot stats; check if they were created in `.addBot`"));
+      // check if bot eats food
+      const food = this.food;
+      const foodIdxToRemove: number[] = [];
+      for (let i = 0; i < food.length; ++i) {
+        const foodElement = food[i];
+        if (!foodElement) {
+          throw new Error("unexpected: food element is undefined");
+        }
+        const distance = World.distance(bot, foodElement);
+        if (distance < bot.radius) {
+          foodIdxToRemove.push(i);
+
+          // update stats
+          const botStats = this.stats.get(bot.botId);
+          if (!botStats) {
+            console.error(new Error("can't find bot stats; check if they were created in `.addBot`"));
+            continue;
+          }
+          botStats.foodEaten += 1;
+        }
+      }
+
+      // remove food and bots
+      for (const botSpawnIdToRemove of botIdsToRemove) {
+        const botToRemove = this.botSpawns.get(botSpawnIdToRemove);
+        if (!botToRemove) {
           continue;
         }
-        botStats.foodEaten += 1;
+        bot.radius += botToRemove.radius;
+        this.botIdToSpawnId.delete(botToRemove.botId);
+        this.botSpawns.delete(botSpawnIdToRemove);
       }
-    }
 
-    // remove food and bots
-    for (const botSpawnIdToRemove of botIdsToRemove) {
-      const botToRemove = botSpawns.get(botSpawnIdToRemove);
-      if (!botToRemove) {
-        continue;
+      // sort in descending order to avoid index shifting when removing elements
+      foodIdxToRemove.sort((a, b) => b - a);
+      for (const foodIdx of foodIdxToRemove) {
+        // it's safe to cast to Sprite because we know that `foodIdx` is within the bounds of the array
+        bot.radius += (food[foodIdx] as Sprite).radius;
+        food.splice(foodIdx, 1);
       }
-      bot.radius += botToRemove.radius;
-      this.botIdToSpawnId.delete(botToRemove.botId);
-      botSpawns.delete(botSpawnIdToRemove);
-    }
 
-    // sort in descending order to avoid index shifting when removing elements
-    foodIdxToRemove.sort((a, b) => b - a);
-    for (const foodIdx of foodIdxToRemove) {
-      // it's safe to cast to Sprite because we know that `foodIdx` is within the bounds of the array
-      bot.radius += (food[foodIdx] as Sprite).radius;
-      food.splice(foodIdx, 1);
+      hasCollisions = hasCollisions || botIdsToRemove.length > 0 || foodIdxToRemove.length > 0;
```

Comment:
First of all, it's better just to return result of expression - function's name tells what exacly it's returning
Secondly, expression has exessive part in it. To explain what i mean, let's see what it really looks like:
```
false || {SOME_VALUE} > 0 || {SOME_OTHER_VALUE} < 0
```
As we can see, it's starts with `false` (value from variable init)

Reply:
@slashnulldot, 

>  it's better just to return result of the expression

It is not because we reset `botIdsToRemove` and `foodIdxToRemove` between the loops 😄 

> Secondly, expression has exessive part in it

No, it doesn't. The `hasCollisions` variable may contain `true` from the previous loop iteration and we need to preserve it 😄

Reply:
@slashnulldot, but your feedback is a good indicator that the `checkCollisions` function is harder to follow. I have simplified it a bit in this PR. Let me know if it's still confusing 😄